File: c_ds/train/basics/rotatingStr.cpp
#include <iostream>
#include <algorithm>

using namespace std;

class Solution {
public:
    bool rotateString(string s, string goal) {
        if (s.length() != goal.length()) return false;
        int slen = s.length();

        for ( int rot_count = 0; rot_count < slen; rot_count++ ) {
            rotate ( s.begin(), s.begin()+1, s.end() );
            if ( s == goal ) return true;
        }
        return false;
    }
};

void test(){
    Solution s;

    {
    string input = "abcdefg";
    string goal = "defgabc";
    cout << input << " -|- " << goal << "\t match status:" << s.rotateString(input, goal) << endl;
    }
    string input = "abcdefg";
    string goal = "d";
    cout << input << " -|- " << goal << "\t match status:" << s.rotateString(input, goal) << endl;
}

int main(){
    test();
    return 0;
}

File: c_ds/train/basics/rotatingStr.py
class Solution:
    def rotateString(self, s: str, goal: str) -> bool:
        slen = len(s)
        if slen == 1:
            return s == goal
        for _ in range(slen):
            s = s[1:] + (s[0])
            if s == goal:
                return True
        return False

def main():
    s = Solution()

    check_str = "abcdefg"
    goal = "defgabc"
    print(s.rotateString(check_str, goal))

    check_str = "abcdefgabcdefg"
    goal = "d"
    print(s.rotateString(check_str, goal))

if __name__ == "__main__":
    main()